scripts/dockerfiles/Dockerfile.main-debug-s3

```diff
@@ -0,0 +1,51 @@
+FROM amazon/aws-for-fluent-bit:main-debug-base
+COPY ./scripts/dockerfiles/Dockerfile.main-debug-base /Dockerfile.2.main-debug-base
+
+ENV S3_BUCKET ""
+ENV S3_KEY_PREFIX issue
+
+RUN yum install -y unzip zip curl gdb
+
+WORKDIR /var/tmp
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-$(arch).zip" -o "awscliv2.zip"
+RUN unzip awscliv2.zip
+RUN ./aws/install
+RUN rm awscliv2.zip
+
+RUN yum remove unzip -y \
+    && yum clean all \
+    && rm -rf /var/cache/yum
+
+# For build efficiency, copy in the Fluent Bit config file again at this stage
+COPY fluent-bit.conf \
+     /fluent-bit/etc/
+
+# Run Fluent Bit from the cores-out folder and collect crash symbols
+# Move symbols to /cores folder after processing
+RUN mkdir /cores-out
+WORKDIR /cores-out
+
+# Only last CMD command will be executed, automatically replaces the original entrypoint
+CMD echo "AWS for Fluent Bit - Debug Image with S3 Core Uploader"; \
+    if [ "$S3_BUCKET" == "" ]; then \
+        echo "Note: Please set S3_BUCKET environment variable to your crash symbol upload destination S3 bucket"; \
+    fi; \
+    if [ "$S3_KEY_PREFIX" == "issue" ]; then \
+        echo "Note: Please set S3_KEY_PREFIX environment variable to a useful identifier - e.g. company name, team name, customer name"; \
+    fi; \
+    export RANDOM_ID_VALUE=$(($RANDOM%99999))$(($RANDOM%99999))$(($RANDOM%99999)); \
+    echo "RANDOM_ID is set to $RANDOM_ID_VALUE"; \
+    /fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.conf; \
+    ls /cores-out/core*; \
+    export CORE_FILENAME=`basename $S3_KEY_PREFIX`_`date +"%FT%H%M%S"`${HOSTNAME+_host-$HOSTNAME}_${RANDOM_ID_VALUE}; \
+    mv `ls /cores-out/core*` ${CORE_FILENAME}.core; \
+    cp /fluent-bit/bin/fluent-bit ./${CORE_FILENAME}.executable; \
+    gdb -batch -ex 'thread apply all bt full' -ex 'quit' '/fluent-bit/bin/fluent-bit' ${CORE_FILENAME}.core > /cores-out/${CORE_FILENAME}.stacktrace \
```

Comment:
Is there any chance this could fail? Or any of the other commands? I'm wondering if we should consider the ordering of these based on what might fail to ensure we get something useful

Reply:
Hmm that's a good thought. Let me update this.

Reply:
I updated the code accordingly and pulled it out into a separate file to reduce code duplication.